view/adminhtml/web/js/import.js

```diff
@@ -42,49 +46,96 @@ define([
                 var self = this;
                 self.element.on('click', function(e) {
                     // Disable the button during the import
-                    $(this).addClass('disabled').attr('disabled', 'disabled').text('Importing orders');
+                    $(this).addClass('disabled').attr('disabled', 'disabled').text('Importing Customers');
+
+                    self.chunkSync(0, self.options.importStatus);
 
-                    if(self.options.totalChunks > 0){
-                        self.options.percentage = (100 / self.options.totalChunks);
-                        self.chunkSync(0);
-                    }
                 });
             },
 
             /**
-             * Post chunk id to proceed orders to Metrilo
+             * Post chunk id to proceed data to Metrilo
              *
              * @param  {integer} chunkId
              * @return {void}
              */
-            chunkSync: function(chunkId) {
+            chunkSync: function(chunkId, importStatus) {
                 var self = this;
                 var progress = Math.round(chunkId * self.options.percentage);
                 self.updateImportingMessage($t('Please wait... ' + progress + '% done'), true);
 
                 var data = {
                     'storeId': self.options.storeId,
                     'chunkId': chunkId,
-                    'totalChunks': self.options.totalChunks,
+                    'customerChunks': self.options.customerChunks,
+                    'categoryChunks': self.options.categoryChunks,
+                    'productChunks': self.options.productChunks,
+                    'orderChunks': self.options.orderChunks,
+                    'importStatus': self.options.importStatus,
                     'form_key': window.FORM_KEY
                 };
-                $.post(self.options.submitUrl, data, function(response) {
-                    if (response.success) {
-                        var newChunkId = chunkId + 1;
-                        if(newChunkId < self.options.totalChunks) {
-                            setTimeout(function() {
-                                self.chunkSync(newChunkId);
-                            }, 100);
-                        } else {
-                            self.updateImportingMessage("<span style='color: green;'>" + $t('Done! Please expect up to 30 minutes for your historical data to appear in Metrilo.') + "</span>");
-                            self.element.removeClass('disabled').addClass('success').text($t('Orders imported'));
-                        }
-                    } else {
-                        self.updateImportingMessage("<span style='color: red;'>" + response.message + "</span>");
+
+                self.ajaxPostWithRetry(self.options.submitUrl, data, 3, function(response) {
+                    var newChunkId = chunkId + 1;
+                    switch (importStatus) {
+                        case 1:
```

Comment:
same question, why not use strings?

Reply:
Edited importStatus to strings of corresponding import type.